modules/tls_mgm/tls_domain.c

```diff
@@ -221,11 +219,138 @@ int set_all_domain_attr(struct tls_domain **dom, char **str_vals, int *int_vals)
 		p = p + eccurve_len + 1;
 	}
 
-	if (pk_len) {
-		d->pkey_file = p;
-		memcpy(p, str_vals[STR_VALS_PK_COL], pk_len);
-		p = p + pk_len + 1;
+	if (blob_vals[BLOB_VALS_PK_COL].len && blob_vals[BLOB_VALS_PK_COL].s) {
+		d->pkey.s = p;
+		d->pkey.len = blob_vals[BLOB_VALS_PK_COL].len;
+		memcpy(p, blob_vals[BLOB_VALS_PK_COL].s, blob_vals[BLOB_VALS_PK_COL].len);
+		p = p + d->pkey.len;
+	}
+
+	return 0;
+}
+
+int set_all_default_domain_attr(struct tls_domain *d, char **str_vals, int *int_vals, str* blob_vals)
```

Comment:
It seems to be a lot of duplicate code with the previous version. Could you say why?